src/number_theory/legendre_symbol/quadratic_char.lean

```diff
@@ -263,14 +273,65 @@ lemma quadratic_char_dichotomy {a : F} (ha : a ≠ 0) :
   quadratic_char F a = 1 ∨ quadratic_char F a = -1 :=
 (sq_eq_one_iff (quadratic_char F a)).mp (quadratic_char_sq_one ha)
 
+/-- For `a : F`, `quadratic_char F a = -1 ↔ ¬ is_square a`. -/
+lemma quadratic_char_neg_one_iff_not_is_square {a : F} :
+  quadratic_char F a = -1 ↔ ¬ is_square a :=
+begin
+  by_cases ha : a = 0,
+  { simp only [ha, is_square_zero, quadratic_char_zero, zero_eq_neg, one_ne_zero, not_true], },
+  { have h : quadratic_char F a = -1 ↔ ¬ quadratic_char F a = 1,
+    { split,
+      { intro h,
+        rw h,
+        norm_num, },
+      { exact λ (h₂ : ¬ quadratic_char F a = 1),
+                (or_iff_right h₂).mp (quadratic_char_dichotomy ha), }, },
```

Comment:
I think this could be a useful lemma in its own right.

Reply:
OK; I will add this: `quadratic_char_eq_neg_one_iff_not_one`?

Reply:
Sounds good to me.